Sources/StreamChatUI/Utils/KeyboardFrameObserver.swift

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+final class KeyboardFrameObserver {
+    weak var containerView: UIView!
+    weak var scrollView: UIScrollView!
+    weak var composerBottomConstraint: NSLayoutConstraint?
+    
+    init(containerView: UIView, scrollView: UIScrollView, composerBottomConstraint: NSLayoutConstraint?) {
+        self.containerView = containerView
+        self.scrollView = scrollView
+        self.composerBottomConstraint = composerBottomConstraint
+    }
```

Comment:
This class takes `containerView`, `scrollView`, and `composerBottomConstraint` as input parameters. It seems to me it's very closely tightened to the message list controllers. Maybe we should reflect it in the name of the class + it should not live in `/Utils`?

Another option would be to more generalize the naming here if you think it's possible and it makes sense.

Reply:
Currently I don't see any reason to make it general. So I renamed it to `ChatMessageListKeyboardObserver` and moved it to `ChatMessage` folder.